static/app/views/performance/browser/webVitals/utils/useTransactionSamplesWebVitalsQuery.tsx

```diff
@@ -0,0 +1,124 @@
+} from 'sentry/views/performance/browser/webVitals/utils/types';
+
+type Props = {
+  transaction: string;
+  limit?: number;
+  orderBy?: WebVitals | null;
+  query?: string;
+};
+
+export const useTransactionSamplesWebVitalsQuery = ({
+  orderBy,
+  limit,
+  transaction,
+  query,
+}: Props) => {
+  const organization = useOrganization();
+  const pageFilters = usePageFilters();
+  const location = useLocation();
+
+  const eventView = EventView.fromNewQueryWithPageFilters(
+    {
+      fields: [
+        'id',
+        'user.display',
+        'transaction',
+        'transaction.op',
+        'measurements.lcp',
+        'measurements.fcp',
+        'measurements.cls',
+        'measurements.ttfb',
+        'measurements.fid',
+        'transaction.duration',
+        'replayId',
+        'timestamp',
+      ],
+      name: 'Web Vitals',
+      query: `transaction.op:pageload transaction:"${transaction}" ${query ? query : ''}`,
+      orderby: mapWebVitalToOrderBy(orderBy),
+      version: 2,
+    },
+    pageFilters.selection
+  );
+
+  const {data, isLoading, ...rest} = useDiscoverQuery({
+    eventView,
+    limit: limit ?? 50,
+    location,
+    orgSlug: organization.slug,
+    options: {
+      enabled: pageFilters.isReady,
+      refetchOnWindowFocus: false,
+    },
+  });
+
+  const tableData: TransactionSampleRowWithScore[] =
+    !isLoading && data?.data.length
+      ? data.data
+          .map(row => ({
+            id: row.id?.toString(),
+            'user.display': row['user.display']?.toString(),
+            transaction: row.transaction?.toString(),
+            'transaction.op': row['transaction.op']?.toString(),
+            'measurements.lcp': row['measurements.lcp'] as number,
+            'measurements.fcp': row['measurements.fcp'] as number,
+            'measurements.cls': row['measurements.cls'] as number,
+            'measurements.ttfb': row['measurements.ttfb'] as number,
+            'measurements.fid': row['measurements.fid'] as number,
+            'transaction.duration': row['transaction.duration'] as number,
+            replayId: row.replayId?.toString(),
+            timestamp: row.timestamp?.toString(),
+          }))
+          .map(row => {
+            const {totalScore, clsScore, fcpScore, lcpScore, ttfbScore, fidScore} =
+              calculatePerformanceScore({
+                lcp: row['measurements.lcp'],
+                fcp: row['measurements.fcp'],
+                cls: row['measurements.cls'],
+                ttfb: row['measurements.ttfb'],
+                fid: row['measurements.fid'],
+              });
+            return {
+              ...row,
+              score: totalScore,
+              clsScore,
+              fcpScore,
+              lcpScore,
+              ttfbScore,
+              fidScore,
+            };
+          })
+      : [];
+
+  return {
+    data: tableData,
+    isLoading,
+    ...rest,
+  };
+};
+
+const mapWebVitalToOrderBy = (webVital?: WebVitals | null) => {
```

Comment:
I think this is duplicated with method in `static/app/views/performance/browser/webVitals/utils/useTransactionWebVitalsQuery.tsx`, but that uses `p75`. Can we extract and make it a single utility?

Reply:
done!